### PR TITLE
Added ability to select default tab

### PR DESCRIPTION
**Pull request by Kardelio**
* Added ability to select default tab

### More detail

Added a ListPreference to act in the settings page
and in the main activity worked with the navGraph to programmatically set the startDestination

Issue #152

### CODE REVIEW

**Comment by Kardelio** [comment]
@mueller-ma let me know what you think <3

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt [hunk 1]
```diff
@@ -45,6 +51,19 @@ class MainActivity : AppCompatActivity() {
         binding.navView.setupWithNavController(navController)
     }
 
+    private fun getDefaultStartDestinationId(defaultTabName: String): Int {
+        return when (defaultTabName) {
+            this.getString(R.string.menu_dices) -> { R.id.nav_dices }
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:56]:
```suggestion
            getString(R.string.menu_dices_value) -> R.id.nav_dices
```

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt [hunk 2]
```diff
@@ -45,6 +51,19 @@ class MainActivity : AppCompatActivity() {
         binding.navView.setupWithNavController(navController)
     }
 
+    private fun getDefaultStartDestinationId(defaultTabName: String): Int {
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:54]: Call `Prefs.defaultTabString()` inside `getDefaultStartDestinationId()`

File: app/src/main/res/values/array.xml [hunk 3]
```diff
@@ -7,4 +7,13 @@
...
     </string-array>
+    <string-array name="tabs">
+        <item>@string/menu_dices</item>
+        <item>@string/menu_random_list</item>
+        <item>@string/menu_rot13</item>
+        <item>@string/menu_alphabet</item>
+        <item>@string/menu_number_converter</item>
+        <item>@string/menu_timer</item>
+        <item>@string/menu_buzzers</item>
+    </string-array>
```
Comment (mueller-ma) [inline app/src/main/res/values/array.xml:18]: These strings may change, e.g. a translation is added. Can you add non-translatable strings by appending `_value`, e.g. `menu_dices_value`?
Then add a second array `tabs_value` and set it as `entryValues` in the `ListPreference`.

**Review by mueller-ma — CHANGES_REQUESTED** [review]
In #152 the pages are called tabs, but IMO they aren't tabs. Can you change it to "Start page"?

File: app/src/main/res/values/array.xml (hunk 3 above)
Comment (Kardelio) [reply app/src/main/res/values/array.xml:18]: Sorry got to be honest im a little bit confused here...
So I should add to `strings.xml` non-translate-able values for each tab title?
And then use that in this `tabs` array?

Sorry little confused here

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt (hunk 2 above)
Comment (Kardelio) [reply app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:54]: Okie

File: app/src/main/res/values/array.xml (hunk 3 above)
Comment (Kardelio) [reply app/src/main/res/values/array.xml:18]: basically one string for the prefs key and one for the display hahaha got it

**Comment by Kardelio** [comment]
@mueller-ma ok done all the requested changes, are they ok?

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt [hunk 4]
```diff
@@ -51,16 +51,16 @@ class MainActivity : AppCompatActivity() {
...
-            this.getString(R.string.menu_dices) -> { R.id.nav_dices }
-            this.getString(R.string.menu_random_list) -> { R.id.nav_random_list }
-            this.getString(R.string.menu_rot13) -> { R.id.nav_rot13 }
-            this.getString(R.string.menu_alphabet) -> { R.id.nav_alphabet }
-            this.getString(R.string.menu_number_converter) -> { R.id.nav_number_converter }
-            this.getString(R.string.menu_timer) -> { R.id.nav_timer }
-            this.getString(R.string.menu_buzzers) -> { R.id.nav_buzzers }
-            else -> { R.id.nav_dices }
+    private fun getDefaultStartDestinationId(): Int {
+        return when (Prefs(context = this).defaultTabString) {
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:55]: You can remove `context = ` here

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -26,6 +26,15 @@
     <string name="number_converter_binary">Binary</string>
     <string name="number_converter_hexadecimal">Hexadecimal</string>
 
+    <!-- Screen / Tab Titles for Default Tab -->
+    <string name="menu_dices_value">dices</string>
```
Comment (mueller-ma) [inline app/src/main/res/values/strings.xml:30]: Needs `translatable="false"` for these strings.

File: app/src/main/res/xml/pref_main.xml [hunk 6]
```diff
@@ -4,7 +4,7 @@
         <ListPreference
                 android:key="default_tab"
                 android:title="@string/default_tab_title"
-                android:entryValues="@array/tabs"
+                android:entryValues="@array/tabs_values"
                 android:entries="@array/tabs"
                 android:defaultValue="@string/menu_dices"
```
Comment (mueller-ma) [inline app/src/main/res/xml/pref_main.xml:9]: `android:defaultValue="@string/menu_dices_value"`

**Review by mueller-ma — CHANGES_REQUESTED** [review]

**Comment by mueller-ma** [comment]
@Kardelio In addition to my review can you replace all user-facing and internal strings from "default tab" to "start page"?

File: app/src/main/res/xml/pref_main.xml (hunk 6 above)
Comment (Kardelio) [reply app/src/main/res/xml/pref_main.xml:9]: Yep

File: app/src/main/res/values/strings.xml (hunk 5 above)
Comment (Kardelio) [reply app/src/main/res/values/strings.xml:30]: yep yep

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt (hunk 4 above)
Comment (Kardelio) [reply app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:55]: okie dokei

**Comment by Kardelio** [comment]
@mueller-ma also i think the Github Actions build needs to be re-run

File: app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt [hunk 7]
```diff
@@ -45,6 +51,19 @@ class MainActivity : AppCompatActivity() {
         binding.navView.setupWithNavController(navController)
     }
 
+    private fun getDefaultStartDestinationId(): Int {
+        return when (Prefs(this).defaultTabString) {
```
Comment (mueller-ma) [inline app/src/main/java/com/github/muellerma/tabletoptools/ui/MainActivity.kt:55]:
```suggestion
        return when (Prefs(this).startPage) {
```

File: app/src/main/res/values/array.xml [hunk 8]
```diff
@@ -7,4 +7,22 @@
         <item>100</item>
         <item>500</item>
     </string-array>
+    <string-array name="tabs">
```
Comment (mueller-ma) [inline app/src/main/res/values/array.xml:10]:
```suggestion
    <string-array name="pages">
```

File: app/src/main/res/values/strings.xml [hunk 9]
```diff
@@ -26,6 +26,15 @@
     <string name="number_converter_binary">Binary</string>
     <string name="number_converter_hexadecimal">Hexadecimal</string>
 
+    <!-- Screen / Tab Titles for Default Page -->
```
Comment (mueller-ma) [inline app/src/main/res/values/strings.xml:29]:
```suggestion
    <!-- Titles for start page -->
```

File: app/src/main/res/values/strings.xml [hunk 10]
```diff
@@ -82,4 +91,5 @@
 
     <!-- Preferences -->
     <string name="menu_preferences">Settings</string>
+    <string name="default_tab_title">Start Page</string>
```
Comment (mueller-ma) [inline app/src/main/res/values/strings.xml:94]:
```suggestion
    <string name="start_page">Start page</string>
```

**Pull request merged**